File: genetic-car-starters/genetic-car-starter-java/src/main/java/fr/genetic/client/java/algo/Random.java
package fr.genetic.client.java.algo;

import java.security.SecureRandom;

/**
 * Created by sebastien on 18/01/2015.
 */
public class Random {

    private static final float CHASSIS_MIN_AXIS = 0.1F;
    private static final float CHASSIS_MAX_AXIS = 1.1F;

    private static final float CHASSIS_MIN_DENSITY = 30F;
    private static final float CHASSIS_MAX_DENSITY = 300F;

    private static final float WHEEL_MIN_RADIUS = 0.2F;
    private static final float WHEEL_MAX_RADIUS = 0.5F;

    private static final float WHEEL_MIN_DENSITY = 40F;
    private static final float WHEEL_MAX_DENSITY = 100F;

    private static final int VERTEX_MAX_VALUE = 7;

    private Random() {

    }

    private static float next(float minValue, float maxValue) {
        return (new SecureRandom().nextFloat() * (maxValue - minValue)) + minValue;
    }

    public static float nextChassisAxis() {
        return next(CHASSIS_MIN_AXIS, CHASSIS_MAX_AXIS);
    }

    public static float nextChassisDensity() {
        return next(CHASSIS_MIN_DENSITY, CHASSIS_MAX_DENSITY);
    }

    public static float nextWheelRadius() {
        return next(WHEEL_MIN_RADIUS, WHEEL_MAX_RADIUS);
    }

    public static float nextWheelDensity() {
        return next(WHEEL_MIN_DENSITY, WHEEL_MAX_DENSITY);
    }

    public static int nextVertex() {
        return new SecureRandom().nextInt(VERTEX_MAX_VALUE + 1);
    }

}

File: genetic-car-starters/genetic-car-starter-csharp/GeneticAlgorithm/Algo/Random.cs
using System;
using System.Security.Cryptography;

namespace GeneticAlgorithm.Algo
{
    public class CustomRandom
    {
        private static RandomNumberGenerator RNG = RandomNumberGenerator.Create();

        private const float CHASSIS_MIN_AXIS = 0.1F;
        private const float CHASSIS_MAX_AXIS = 1.1F;

        private const float CHASSIS_MIN_DENSITY = 30F;
        private const float CHASSIS_MAX_DENSITY = 300F;

        private const float WHEEL_MIN_RADIUS = 0.2F;
        private const float WHEEL_MAX_RADIUS = 0.5F;

        private const float WHEEL_MIN_DENSITY = 40F;
        private const float WHEEL_MAX_DENSITY = 100F;

        private const int VERTEX_MAX_VALUE = 7;

        private CustomRandom()
        {
        }

        private static double NextDouble()
        {
            byte[] b = new byte[4];
            RNG.GetBytes(b);
            return (double)BitConverter.ToUInt32(b, 0) / UInt32.MaxValue;
        }

        private static float Next(float minValue, float maxValue)
        {
            return (float) (NextDouble() * (maxValue - minValue) + minValue);
        }

        public static float NextChassisAxis()
        {
            return Next(CHASSIS_MIN_AXIS, CHASSIS_MAX_AXIS);
        }

        public static float NextChassisDensity()
        {
            return Next(CHASSIS_MIN_DENSITY, CHASSIS_MAX_DENSITY);
        }

        public static float NextWheelRadius()
        {
            return Next(WHEEL_MIN_RADIUS, WHEEL_MAX_RADIUS);
        }

        public static float NextWheelDensity()
        {
            return Next(WHEEL_MIN_DENSITY, WHEEL_MAX_DENSITY);
        }

        public static int NextVertex()
        {
            return (int) Next(0, VERTEX_MAX_VALUE + 1);
        }
    }
}
